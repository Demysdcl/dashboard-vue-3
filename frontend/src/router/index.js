import Home from '@/views/Home'
import { createRouter, createWebHistory } from 'vue-router'

const createRoute = (name, hasAuth = true) => ({
  path: `/${name.toLowerCase()}`,
  name,
  meta: { hasAuth },
  component: () =>
    import(/* webpackChunkName: "[request]" */ `../views/${name}/index.vue`),
})

const views = () => ['Feedbacks', 'Credentials'].map(createRoute)

export const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home,
  },
  ...views(),
]

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes,
})

export default router
