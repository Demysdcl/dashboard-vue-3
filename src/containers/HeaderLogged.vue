<script>
import Link from '@/components/Link'
import { computed, reactive, toRefs } from 'vue'
import { useRouter } from 'vue-router'
import useStore from '@/hooks/useStore'
import { cleanCurrentUser } from '@/store/user'

export default {
  components: {
    Link,
  },
  setup() {
    const store = useStore('User')
    const state = reactive({
      links: [
        {
          title: 'Credenciais',
          to: 'Credentials',
        },
        {
          title: 'Feedbacks',
          to: 'Feedbacks',
        },
      ],
    })

    const it = {
      ...toRefs(state),
      router: useRouter(),
      logoutLabel: computed(() =>
        store.currentUser.name ? `${store.currentUser.name} (sair)` : '...',
      ),
      handleLogout() {
        localStorage.removeItem('token')
        cleanCurrentUser()
        it.router.push({ name: 'Home' })
      },
    }
    return it
  },
}
</script>

<template>
  <div class="flex items-center justify-between w-4/5 max-w-6xl py-10">
    <div class="w-28 lg:w-36">
      <img class="w-full" alt="logo" src="../assets/images/logo_white.png" />
    </div>

    <div class="flex">
      <ul class="flex list-none">
        <Link
          v-for="link in links"
          :title="link.title"
          :to="link.to"
          :key="link.title"
        />

        <li
          @click="handleLogout"
          class="px-6 py-2 font-bold bg-white rounded-full cursor-pointer text-brand-main focus:outline-none"
        >
          {{ logoutLabel }}
        </li>
      </ul>
    </div>
  </div>
</template>
