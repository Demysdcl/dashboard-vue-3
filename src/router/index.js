import { createRouter, createWebHistory } from 'vue-router'
import Home from '@/views/Home'

const createRoute = (name) => {
  return {
    path: `/${name.toLowerCase()}`,
    name,
    component: () =>
      import(/* webpackChunkName: "[request]" */ `../views/${name}/index.vue`),
  }
}

const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home,
  },
  createRoute('Feedbacks'),
]

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes,
})

export default router
