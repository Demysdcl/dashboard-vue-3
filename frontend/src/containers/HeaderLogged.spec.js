import { routes } from '@/router'
import { shallowMount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'
import HeaderLogged from './HeaderLogged'

const router = createRouter({
  history: createWebHistory(),
  routes: routes,
})

const mockStore = { currentUser: {} }
jest.mock('@/hooks/useStore', () => {
  return () => {
    return mockStore
  }
})

describe('<HeaderLogged />', () => {
  const wrapper = shallowMount(HeaderLogged, {
    global: { plugins: [router] },
  })

  beforeEach(async () => {
    router.push('/')
    await router.isReady()
  })

  it('should render HeaderLogged', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render 3 dots when there isnÂ´t user logged', () => {
    const logoutButton = wrapper.find('#logout-button')
    expect(logoutButton.text()).toBe('...')
  })

  it('should render user name when there`s user logged', () => {
    mockStore.currentUser.name = 'Demys'
    const wrapper = shallowMount(HeaderLogged, {
      global: { plugins: [router] },
    })
    const logoutButton = wrapper.find('#logout-button')
    expect(logoutButton.text()).toBe('Demys (sair)')
  })
})
