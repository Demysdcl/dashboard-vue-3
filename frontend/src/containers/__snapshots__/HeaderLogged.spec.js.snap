// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLogged /> should render HeaderLogged 1`] = `
<div class="flex items-center justify-between w-4/5 max-w-6xl py-10">
  <div class="w-28 lg:w-36"><img class="w-full" alt="logo" src=""></div>
  <div class="flex">
    <ul class="flex list-none">
      <link-stub title="Credenciais" to="Credentials"></link-stub>
      <link-stub title="Feedbacks" to="Feedbacks"></link-stub>
      <li id="logout-button" class="px-6 py-2 font-bold bg-white rounded-full cursor-pointer text-brand-main focus:outline-none">...</li>
    </ul>
  </div>
</div>
`;
